libx264/ffmpeg.1
.TH FFMPEG 1  "ffmpeg 0.0.0" 
.SH NAME
ffmpeg
.SH SYNOPSIS
\fBffmpeg\fR [OPTIONS] [OUTFILE]...
.SH OPTIONS
.TP
\fB\-L\fR
.br
show license
.TP
\fB\-h\fR=\fITOPIC\fR
.br
show help
.TP
\fB\-version\fR
.br
show version
.TP
\fB\-muxers\fR
.br
show available muxers
.TP
\fB\-demuxers\fR
.br
show available demuxers
.TP
\fB\-devices\fR
.br
show available devices
.TP
\fB\-decoders\fR
.br
show available decoders
.TP
\fB\-encoders\fR
.br
show available encoders
.TP
\fB\-filters\fR
.br
show available filters
.TP
\fB\-pix_fmts\fR
.br
show available pixel formats
.TP
\fB\-layouts\fR
.br
show standard channel layouts
.TP
\fB\-sample_fmts\fR
.br
show available audio sample formats
.TP
\fB\-v\fR=\fILOGLEVEL\fR
.br
set logging level
.TP
\fB\-y\fR
.br
overwrite output files
.TP
\fB\-n\fR
.br
never overwrite output files
.TP
\fB\-print_graphs\fR
.br
print execution graph data to stderr
.TP
\fB\-print_graphs_file\fR=\fIFILENAME\fR
.br
write execution graph data to the specified file
.TP
\fB\-print_graphs_format\fR=\fIFORMAT\fR
.br
set the output printing format
.br

.br
[\fIpossible values: \fRdefault, compact, csv, flat, ini, json, xml, mermaid, mermaidhtml]
.TP
\fB\-stats\fR
.br
print progress report during encoding
.TP
\fB\-f\fR=\fIFMT\fR
.br
force container format (auto\-detected otherwise)
.TP
\fB\-t\fR=\fIDURATION\fR
.br
stop transcoding after specified duration
.TP
\fB\-to\fR=\fITIME\-STOP\fR
.br
stop transcoding after specified time is reached
.TP
\fB\-ss\fR=\fITIME\-OFF\fR
.br
start transcoding at specified time
.TP
\fB\-metadata\fR \fI:<SPEC>\fR \fIKEY=VALUE\fR
.br
add metadata
.TP
\fB\-c\fR \fI:<STREAM\-SPEC>\fR \fICODEC\fR
.br
select encoder/decoder ('copy' to copy stream without reencoding)
.TP
\fB\-filter\fR \fI:<STREAM\-SPEC>\fR \fIFILTER\-GRAPH\fR
.br
apply specified filters to audio/video
.TP
\fB\-r\fR \fI:<STREAM\-SPEC>\fR \fIRATE\fR
.br
override input framerate/convert to given output framerate (Hz value, fraction or abbreviation)
.TP
\fB\-aspect\fR \fI:<STREAM\-SPEC>\fR \fIASPECT\fR
.br
set aspect ratio (4:3, 16:9 or 1.3333, 1.7777)
.TP
\fB\-vn\fR
.br
disable video
.TP
\fB\-vcodec\fR=\fICODEC\fR
.br
alias for \-c:v (select encoder/decoder for video streams)
.TP
\fB\-vf\fR=[\fIFILTER\-GRAPH\fR]...
.br
alias for \-filter:v (apply filters to video streams)
.TP
\fB\-b\fR=\fIBITRATE\fR
.br
video bitrate (please use \-b:v)
.TP
\fB\-aq\fR=\fIQUALITY\fR
.br
set audio quality (codec\-specific)
.TP
\fB\-ar\fR \fI:<STREAM\-SPEC>\fR \fIRATE\fR
.br
set audio sampling rate (in Hz)
.br

.br
[\fIpossible values: \fR22050, 44100, 48000]
.TP
\fB\-ac\fR \fI:<STREAM\-SPEC>\fR \fICHANNELS\fR
.br
set number of audio channels
.TP
\fB\-an\fR
.br
disable audio
.TP
\fB\-acodec\fR=\fICODEC\fR
.br
alias for \-c:a (select encoder/decoder for audio streams)
.TP
\fB\-ab\fR=\fIBITRATE\fR
.br
alias for \-b:a (select bitrate for audio streams)
.br

.br
[\fIpossible values: \fR16, 32, 64, 128, 192, 256, 320]
.TP
\fB\-af\fR=[\fIFILTER\-GRAPH\fR]...
.br
alias for \-filter:a (apply filters to audio streams)
.TP
\fB\-sn\fR
.br
disable subtitle
.TP
\fB\-scodec\fR=\fICODEC\fR
.br
alias for \-c:s (select encoder/decoder for subtitle streams)
.TP
\fB\-help\fR
.br
Print help
.TP
[\fIOUTFILE\fR]...



completions/ffmpeg.sh
#!/usr/bin/env bash
# Automatic generated, DON'T MODIFY IT.

# @flag -L                                     show license
# @option -h[`_choice_help_topic`] <topic>     show help
# @flag -version                               show version
# @flag -muxers                                show available muxers
# @flag -demuxers                              show available demuxers
# @flag -devices                               show available devices
# @flag -decoders                              show available decoders
# @flag -encoders                              show available encoders
# @flag -filters                               show available filters
# @flag -pix_fmts                              show available pixel formats
# @flag -layouts                               show standard channel layouts
# @flag -sample_fmts                           show available audio sample formats
# @option -v[`_choice_loglevel`] <loglevel>    set logging level
# @flag -y                                     overwrite output files
# @flag -n                                     never overwrite output files
# @flag -stats                                 print progress report during encoding
# @option -f[`_choice_format`] <fmt>           force container format (auto-detected otherwise)
# @option -t <duration>                        stop transcoding after specified duration
# @option -to <time_stop>                      stop transcoding after specified time is reached
# @option -ss <time_off>                       start transcoding at specified time
# @option -metadata <:<spec>> <key=value>      add metadata
# @option -c[`_choice_codec`] <:<stream_spec>> <codec>  select encoder/decoder ('copy' to copy stream without reencoding)
# @option -filter <:<stream_spec>> <filter_graph>  apply specified filters to audio/video
# @option -r <:<stream_spec>> <rate>           override input framerate/convert to given output framerate (Hz value, fraction or abbreviation)
# @option -aspect <:<stream_spec>> <aspect>    set aspect ratio (4:3, 16:9 or 1.3333, 1.7777)
# @flag -vn                                    disable video
# @option -vcodec[`_choice_codec`] <codec>     alias for -c:v (select encoder/decoder for video streams)
# @option -vf*[`_choice_filter_kv`] <filter_graph>  alias for -filter:v (apply filters to video streams)
# @option -b <bitrate>                         video bitrate (please use -b:v)
# @option -aq <quality>                        set audio quality (codec-specific)
# @option -ar[22050|44100|48000] <:<stream_spec>> <rate>  set audio sampling rate (in Hz)
# @option -ac <:<stream_spec>> <channels>      set number of audio channels
# @flag -an                                    disable audio
# @option -acodec[`_choice_codec`] <codec>     alias for -c:a (select encoder/decoder for audio streams)
# @option -ab[16|32|64|128|192|256|320] <bitrate>  alias for -b:a (select bitrate for audio streams)
# @option -af*[`_choice_filter_kv`] <filter_graph>  alias for -filter:a (apply filters to audio streams)
# @flag -sn                                    disable subtitle
# @option -scodec[`_choice_codec`] <codec>     alias for -c:s (select encoder/decoder for subtitle streams)
# @arg outfile*

. "$ARGC_COMPLETIONS_ROOT/utils/_argc_utils.sh"

_choice_help_topic() {
    cat <<-'EOF' | _argc_util_comp_kv =
long;;Print advanced tool options in addition to the basic tool options.
full;;Print complete list of options
decoder=`_choice_decoder`;;Print detailed information about the decoder
encoder=`_choice_encoder`;;Print detailed information about the encoder
demuxer=`_choice_demuxer`;;Print detailed information about the demuxer
muxer=`_choice_muxer`;;Print detailed information about the muxer
filter=`_choice_filter_key`;;Print detailed information about the filter
bsf=`_choice_bsf`;;Print detailed information about the bitstream filter
protocol=`_choice_protocol`;;Print detailed information about the protocol
EOF
}

_choice_loglevel() {
    cat <<-'EOF'
quiet	Show nothing at all; be silent.
panic	Only show fatal errors which could lead the process to crash
fatal	Only show fatal errors.
error	Show all errors, including ones which can be recovered from.
warning	Show all warnings and errors.
info	Show informative messages during processing.
verbose	Same as "info", except more verbose.
debug	Show everything, including debugging information.
trace	
EOF
}

_choice_format() {
    ffmpeg -hide_banner -formats | tail -n +5 | sed 's/^.\{3\} \([^ ]\+\) \(.*\)$/\1\t\2/'
}

_choice_codec() {
    ffmpeg -hide_banner -codecs | tail -n +11 | sed 's/^.\{7\} \([^ ]\+\) \(.*\)$/\1\t\2/'
}

_choice_filter_kv() {
    _argc_util_mode_kv =
    _choice_filter | _argc_util_transform suffix== nospace
}

_choice_bsf() {
    ffmpeg -hide_banner -bsfs | tail -n +2
}

_choice_decoder() {
    ffmpeg -hide_banner -decoders | tail -n +11 | sed 's/^.\{7\} \([^ ]\+\) \(.*\)$/\1\t\2/'
}

_choice_demuxer() {
    ffmpeg -hide_banner -demuxers | tail -n +5 | sed 's/^.\{3\} \([^ ]\+\) \(.*\)$/\1\t\2/'
}

_choice_encoder() {
    ffmpeg -hide_banner -encoders | tail -n +11 | sed 's/^.\{7\} \([^ ]\+\) \(.*\)$/\1\t\2/'
}

_choice_filter() {
    ffmpeg -hide_banner -filters | tail -n +9 | sed 's/^.\{4\} \([^ ]\+\) \+[^ ]\+ \(.*\)$/\1\t\2/'
}

_choice_muxer() {
    ffmpeg -hide_banner -muxers | tail -n +5 | sed 's/^.\{3\} \([^ ]\+\) \(.*\)$/\1\t\2/'
}

_choice_protocol() {
    ffmpeg -hide_banner -protocols | tail -n +3
}

command eval "$(argc --argc-eval "$0" "$@")"